File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/ParameterValidation.java
package com.attentive.androidsdk;

import android.util.Log;
import java.util.Collection;

// TODO move to 'internal' package
public class ParameterValidation {
    public static boolean verifyNotNull(Object param, String paramName) {
        if (param == null) {
            Log.e(ParameterValidation.class.getName(), paramName + " cannot be null.");
            return false;
        }
        return true;
    }

    public static boolean verifyNotEmpty(String param, String paramName) {
        if (verifyNotNull(param, paramName) && param.isEmpty()) {
            Log.e(ParameterValidation.class.getName(), paramName + " cannot be empty.");
            return false;
        }
        return true;
    }

    public static <T> boolean verifyNotEmpty(Collection<T> param, String paramName) {
        if (verifyNotNull(param, paramName) && param.isEmpty()) {
            Log.e(ParameterValidation.class.getName(), paramName + " cannot be empty.");
            return false;
        }
        return true;
    }
}

File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/ParameterValidation.kt
package com.attentive.androidsdk

// TODO move to 'internal' package
object ParameterValidation {
    @JvmStatic
    fun verifyNotNull(param: Any?, paramName: String) {
        requireNotNull(param) { "$paramName cannot be null." }
    }

    @JvmStatic
    fun verifyNotEmpty(param: String, paramName: String) {
        verifyNotNull(param, paramName)

        require(param.isNotEmpty()) { "$paramName cannot be empty." }
    }

    @JvmStatic
    fun verifyNotEmpty(param: Collection<*>, paramName: String) {
        verifyNotNull(param, paramName)

        require(!param.isEmpty()) { "$paramName cannot be empty." }
    }
}
